tests/basic_correctness/test_preemption.py

```diff
@@ -0,0 +1,138 @@
+"""Compare the short outputs of HF and vLLM when using greedy sampling.
+
+VLLM_TEST_ENABLE_ARTIFICIAL_PREEMPT=1 has to be set before running this test.
+
+Run `VLLM_TEST_ENABLE_ARTIFICIAL_PREEMPT=1
+pytest tests/basic_correctness/test_preemption.py`.
+"""
+import pytest
+
+from vllm.core.scheduler import (ARTIFICIAL_PREEMPTION_MAX_CNT,
+                                 ENABLE_ARTIFICIAL_PREEMPT)
+
+MODELS = [
+    "facebook/opt-125m",
+]
+
+assert ENABLE_ARTIFICIAL_PREEMPT is True, (
+    "Use an env var VLLM_TEST_ENABLE_ARTIFICIAL_PREEMPT=1. "
+    "`VLLM_TEST_ENABLE_ARTIFICIAL_PREEMPT=1 pytest "
+    "tests/basic_correctness/test_preemption.py`")
```

Comment:
I created `num_gpu_blocks_override` for spec decode preemption tests and it works pretty well!

https://github.com/vllm-project/vllm/blob/df29793dc73a83f3c86c19de967adffda1a28a93/tests/spec_decode/e2e/test_correctness.py#L351-L395

Reply:
How is `2 + 256 // 8,` decided? one concern is that it could be a little prone to code change (e.g., example). Or is there a way to verify if preemption happens from the engine now?

Reply:
Okay, followed the suggestion and added test-only var to scheduler to verify if preemption happens